src/pydocstyle/config.py

```diff
@@ -23,6 +25,98 @@ def _decorator(self, *args, **kwargs):
     return _decorator
 
 
+class TomlParser:
+    """ConfigParser that partially mimics RawConfigParser but for toml files.
+
+    See RawConfigParser for more info. Also, please note that not all
+    RawConfigParser functionality is implemented, but only the subset, that is
+    currently used by pydocstyle.
+    """
+
+    def __init__(self):
+        """Create a toml parser."""
+        self._config = {}
+
+    def read(self, filenames, encoding=None):
+        """Read and parse a filename or an iterable of filenames.
+
+        Files that cannot be opened are silently ignored; this is
+        designed so that you can specify an iterable of potential
+        configuration file locations (e.g. current directory, user's
+        home directory, systemwide directory), and all existing
+        configuration files in the iterable will be read.  A single
+        filename may also be given.
+
+        Return list of successfully read files.
+        """
+        if isinstance(filenames, (str, bytes, os.PathLike)):
+            filenames = [filenames]
+        read_ok = []
+        for filename in filenames:
+            try:
+                with open(filename, encoding=encoding) as fp:
+                    self._config.update(toml.load(fp))
+            except OSError:
+                continue
+            if isinstance(filename, os.PathLike):
+                filename = os.fspath(filename)
+            read_ok.append(filename)
+        return read_ok
+
+    def _get_section(self, section, allow_none=False):
+        current = self._config
+        for p in section.split('.'):
+            if isinstance(current, dict) and p in current:
+                current = current[p]
+            else:
+                current = None
+                break
```

Comment:
```suggestion
        try:
            current = reduce(operator.getitem, section.split('.'), self._config)
        except KeyError:
            current = None
```
Using `operator.getitem` and `functools.reduce` just seems cleaner to me. Less verbose, easier to read. Personal preference on that one though.